examples/dreambooth/train_dreambooth_lora_hidream.py

```diff
@@ -0,0 +1,1829 @@
+        initial_global_step = 0
+
+    progress_bar = tqdm(
+        range(0, args.max_train_steps),
+        initial=initial_global_step,
+        desc="Steps",
+        # Only show the progress bar once on each machine.
+        disable=not accelerator.is_local_main_process,
+    )
+
+    def get_sigmas(timesteps, n_dim=4, dtype=torch.float32):
+        sigmas = noise_scheduler_copy.sigmas.to(device=accelerator.device, dtype=dtype)
+        schedule_timesteps = noise_scheduler_copy.timesteps.to(accelerator.device)
+        timesteps = timesteps.to(accelerator.device)
+        step_indices = [(schedule_timesteps == t).nonzero().item() for t in timesteps]
+
+        sigma = sigmas[step_indices].flatten()
+        while len(sigma.shape) < n_dim:
+            sigma = sigma.unsqueeze(-1)
+        return sigma
+
+    for epoch in range(first_epoch, args.num_train_epochs):
+        transformer.train()
+
+        for step, batch in enumerate(train_dataloader):
+            models_to_accumulate = [transformer]
+            prompts = batch["prompts"]
+
+            with accelerator.accumulate(models_to_accumulate):
+                # encode batch prompts when custom prompts are provided for each image -
+                if train_dataset.custom_instance_prompts:
+                    prompt_embeds, pooled_prompt_embeds = compute_text_embeddings(prompts, text_encoders, tokenizers)
+
+                # Convert images to latent space
+                if args.cache_latents:
+                    model_input = latents_cache[step].sample()
+                else:
+                    vae = vae.to(accelerator.device)
+                    pixel_values = batch["pixel_values"].to(dtype=vae.dtype)
+                    model_input = vae.encode(pixel_values).latent_dist.sample()
+                    if args.offload:
+                        vae = vae.to("cpu")
+                model_input = (model_input - vae_config_shift_factor) * vae_config_scaling_factor
+                model_input = model_input.to(dtype=weight_dtype)
+
+                if model_input.shape[-2] != model_input.shape[-1]:
+                    B, C, H, W = model_input.shape
+                    pH, pW = H // transformer.config.patch_size, W // transformer.config.patch_size
+
+                    img_sizes = torch.tensor([pH, pW], dtype=torch.int64).reshape(-1)
+                    img_ids = torch.zeros(pH, pW, 3)
+                    img_ids[..., 1] = img_ids[..., 1] + torch.arange(pH)[:, None]
+                    img_ids[..., 2] = img_ids[..., 2] + torch.arange(pW)[None, :]
+                    img_ids = img_ids.reshape(pH * pW, -1)
+                    img_ids_pad = torch.zeros(self.transformer.max_seq, 3)
+                    img_ids_pad[: pH * pW, :] = img_ids
+
+                    img_sizes = img_sizes.unsqueeze(0).to(model_input.device)
+                    img_ids = img_ids_pad.unsqueeze(0).to(model_input.device)
+
+                else:
+                    img_sizes = img_ids = None
```

Comment:
Would add a comment explaining what is happening and move the `img_ids` calculation in a function.

Reply:
This was moved into the `transformer` so should be ok to remove

https://github.com/huggingface/diffusers/blob/a8f5134c113da402a93580ef7a021557e816c98d/src/diffusers/models/transformers/transformer_hidream_image.py#L743-L756